src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,70 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_CAPACITY = 10;
+    private Pair<K, V>[] pairs;
+
+    private static class Pair<K, V> {
+        private K key;
+        private V value;
+
+        private Pair(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        private K getKey() {
+            return key;
+        }
+
+        private V getValue() {
+            return value;
+        }
+
+        private void setValue(V value) {
+            this.value = value;
+        }
+    }
+
+    public StorageImpl() {
+        pairs = new Pair[STORAGE_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < pairs.length; i++) {
+            if (pairs[i] != null) {
+                if (pairs[i].getKey() == null ? key == null : pairs[i].getKey().equals(key)) {
+                    pairs[i].setValue(value);
+                    break;
+                }
+            } else {
+                pairs[i] = new Pair<>(key, value);
+                break;
+            }
+        }
     }
 
     @Override
     public V get(K key) {
+        for (Pair<K, V> pair : pairs) {
+            if (pair != null) {
+                if (pair.getKey() == null ? key == null : pair.getKey().equals(key)) {
+                    return pair.getValue();
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        int result = 0;
+        for (Pair<K, V> pair : pairs) {
+            if (pair != null) {
+                result++;
+            }
+        }
+        return result;
```

Comment:
Please introduce class variable and increment it after adding new element in order not to calculate size each time by iterating
```suggestion
        return size;
```